dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/execute/AsyncMasterTaskDelayQueueLooper.java

```diff
@@ -86,8 +88,14 @@ public void run() {
                     try {
                         LogUtils.setTaskInstanceLogFullPathMDC(taskExecutionContext.getLogPath());
                         LogUtils.setTaskInstanceIdMDC(taskExecutionContext.getTaskInstanceId());
-                        AsyncTaskExecuteFunction.AsyncTaskExecutionStatus asyncTaskExecutionStatus =
-                                asyncTaskExecuteFunction.getAsyncTaskExecutionStatus();
+                        AsyncTaskExecuteFunction.AsyncTaskExecutionStatus asyncTaskExecutionStatus;
+                        if (taskExecutionContext.getDryRun() == DRY_RUN_FLAG_YES) {
+                            log.info(
+                                    "The current execute mode is dry run check again, will stop the logic task and set the taskInstance status to success");
+                            asyncTaskExecutionStatus = AsyncTaskExecuteFunction.AsyncTaskExecutionStatus.SUCCESS;
+                        } else {
+                            asyncTaskExecutionStatus = asyncTaskExecuteFunction.getAsyncTaskExecutionStatus();
+                        }
```

Comment:
We can revert this change, since the Async task will not put into asyncMasterTaskDelayQueue

Reply:
> We can revert this change, since the Async task will not put into asyncMasterTaskDelayQueue

This is for code robustness considerations. I'm not sure if there are other places where events can be sent.